core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -55,7 +59,50 @@ public class RESTCatalog
   public RESTCatalog() {
     this(
         SessionCatalog.SessionContext.createEmpty(),
-        config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build());
+        config -> {
+          HTTPClient.Builder builder =
+              HTTPClient.builder(config).uri(config.get(CatalogProperties.URI));
+
+          String proxyHostname =
+              PropertyUtil.propertyAsString(
+                  config,
+                  CatalogProperties.PROXY_HOSTNAME,
+                  CatalogProperties.PROXY_HOSTNAME_DEFAULT);
+
+          int proxyPort =
+              PropertyUtil.propertyAsInt(
+                  config, CatalogProperties.PROXY_PORT, CatalogProperties.PROXY_PORT_DEFAULT);
```

Comment:
Minor: move it into the if clause(line 77)?

Reply:
I'm not sure we should be defaulting to a proxy port. I'd rather mandate that the port always needs to be provided, hence this would be `Integer proxyPort = PropertyUtil.propertyAsNullableInt(..)`. And then later you would null check both the hostname/port to determine whether `builder.withProxy(proxyHostname, proxyPort)` can be called